src/argilla/client/nightly.py

```diff
@@ -0,0 +1,231 @@
+                                response=record["response"] if "response" in record else {"values": {}},
+                                external_id=record["external_id"],
+                                inserted_at=record["inserted_at"],
+                                updated_at=record["inserted_at"],
+                            )
+                            for record in self.client.get_records(
+                                id=self.id, offset=prev_limit, limit=prev_limit + increment
+                            ).parsed["items"]
+                        ]
+                        pbar.update(increment)
+        return self.__records
+
+    @property
+    def questions(self) -> List[QuestionSchema]:
+        return self.__questions
+
+    def __repr__(self) -> str:
+        return (
+            f"Dataset(name={self.name}, workspace={self.workspace}, schema={self.schema},"
+            f" description={self.description}, guidelines={self.guidelines}, questions={self.questions},"
+            f" status={self.status})"
+        )
+
+    def add_record(
+        self,
+        record: Union[FieldsSchema, Dict[str, Any]],
+        response: Optional[Dict[str, Any]] = {"values": {}},
+        external_id: Optional[str] = None,
+    ) -> None:
+        if self.status is DatasetStatus.DRAFT:
+            raise ValueError("Cannot add records to a dataset that is in draft status, please publish it first.")
+        if self.schema is None:
+            warnings.warn("Since the `schema` hasn't been defined during the dataset creation, it will be inferred.")
+            self.schema = generate_pydantic_schema(record)
+        # # If there are records already logged to Argilla, fetch one and get the schema
+        # self.schema = generate_pydantic_schema(self.fetch_one())
+        # # If there are no records logged to Argilla, check if `self.schema` has been set
+        # ...
+        # # If `self.schema` has not been set, just infer the schema based on the record
+        # ...
+        # record = record.dict() if isinstance(record, FieldsSchema) else record
+        self.client.add_record(
+            id=self.id,
+            record=RecordSchema(fields=self.schema(**record), external_id=external_id, response=response).dict(),
+        )
+        if self.__records is not None and isinstance(self.__records, list) and len(self.__records) > 0:
+            self.records.append(self.schema(**record))
+
+    def fetch_one(self) -> Union[Dict[str, Any], List[str, Any]]:
+        if self.__records is None or len(self.__records) < 1:
+            # TODO: handle exception if there are no records
+            return self.client.get_records(id=self.id, offset=0, limit=1).parsed["items"][0]
+        return self.__records[0]
+
+    # TODO: we could fetch those on iter, maybe we can create an `streaming` flag or something similar
+    def iter(self, batch_size: int = 32) -> List[BaseModel]:
```

Comment:
Here I'll probably create another class named `StreamingDataset` to stream the data during training if that's something we need/want, but still need to discuss it